### PR TITLE
Add coding style page

### PR DESCRIPTION
**Pull request by zaibon**
Describe the coding convention we want to enforce to our code repos

### CODE REVIEW

File: CodingCycle.md [hunk 1]
```diff
@@ -1,51 +1,48 @@
...
-
-### Step 5b: Someone needs your code
-
-- If other stories depend on your code, intermediate pull requests can be created, validated and merged in on the other branch or master (careful!)
-- Best to avoid this step if not required, normally it means your story was too big and should have been cut in multiple pieces
-
+### Step 5: code reviews
+- You're code will be reviewed by your peer.
+- disscusion can happen on the pull request to see how to solve best a problem or if someone has a better solution to the problem.
+- restat the cycle from step 3 to step 5 till everyone agree on the code produced.
```
Comment (grimpy) [inline CodingCycle.md]: restart*

Comment (zaibon) [reply CodingCycle.md]: yep, and two steps 5 too

File: CodingCycle.md [hunk 2]
```diff
@@ -1,51 +1,48 @@
...
+- write the documentation of the feature you implement
 
-### Step 4: Commit / code reviews
+### Step 4: Commit
 
-- Perform a peer review before you commit. The goal is that you go over your diff while explaining to your peer what you changed. It happens very often that you find bugs or shortcomings while you are explaining your work.
-- Include the reference to the task in your commit
-   - Notation format storyname: #taskreference
-   - E.g. "Implemented live migration cockpit: #10"
+- Once your code is ready, the test a written and the documentation is good enough, push your local branch to remote and create a pull request. Your pull request should ask to merge your branch to the version branch, never to master. (check [branching strategy](prodorg_branching_strategies.md) for more detail)
```
Comment (rkhamis) [inline CodingCycle.md]: the tests are written*

File: CodingCycle.md [hunk 3]
```diff
@@ -1,51 +1,48 @@
...
+- Include the reference to the issue you're solving in your commit
+  - E.g. "Implemented live migration cockpit: #10"
 
+### Step 5: Auto test perfome by CI
+- If the project you're working on has a CI enable, make sure that all the test passes and that your branch has not conflict.
```
Comment (rkhamis) [inline CodingCycle.md]: *has CI enabled.
*your branch doesn't have conflicts

File: CodingCycle.md [hunk 4]
```diff
@@ -1,51 +1,48 @@
...
+- People won't even start reviewing your code if the tests don't pass or if your branch is out of date.
 
-### Step 5: Test / code reviews
+### Step 6: code reviews
+- You're code will be reviewed by your peer.
```
Comment (rkhamis) [inline CodingCycle.md]: *Your

File: CodingStyle.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+- For Go Always run `gofmt -s -w file.go` on each changed file before committing your changes. Most editors have plug-ins that do this automatically.  
+For Atom: https://github.com/joefitzgerald/gofmt
+
+- For Python we choose to follow the [PEP8 style guide](https://www.python.org/dev/peps/pep-0008/)  
+Always run `autopep8 -a --max-length 120 file.py` on each changed file before committing your changes. Most editors have plug-ins that do this automatically.  
+For Atom: https://github.com/markbaas/atom-python-autopep8
+
+
+
+### Pull request
```
Comment (rkhamis) [inline CodingStyle.md:29]: nice!

**Review by despiegk — APPROVED** [review]

**Pull request merged**